src/calls.c

```diff
@@ -1543,12 +1533,27 @@ Obj FuncCALL_FUNC_LIST (
         result = CALL_XARGS( func, list2 );
       }
     } else {
-      result = DoOperation2Args(self, func, list);
+      result = DoOperation2Args(CallFuncListOper, func, list);
     }
     /* return the result                                                   */
     return result;
+
 }
 
+Obj FuncCALL_FUNC_LIST (
+    Obj                 self,
+    Obj                 func,
+    Obj                 list )
+{
+  /* check that the second argument is a list                            */
+    while ( ! IS_SMALL_LIST( list ) ) {
+        list = ErrorReturnObj(
+            "CallFuncList: <list> must be a small list",
+            0L, 0L,
+            "you can replace <list> via 'return <list>;'" );
+    }
+    return CallFuncList(func, list);
+}
 
```

Comment:
Breaking out `CallFuncList` from `FuncCALL_FUNC_LIST` seems like a very good idea, though it would of course be nicer if this was done in a separate commit, instead of mingling it with the change that is actually described in the commit message. At the least, I'd expect the commit message to mention this change.